frontend/src/Editor/Components/Table/Table.jsx

```diff
@@ -783,6 +792,25 @@ export function Table({
     }
   }, [JSON.stringify(changeSet)]);
 
+  useEffect(() => {
+    if (allowSelection && typeof defaultSelectedRow === 'object' && !_.isEmpty(defaultSelectedRow)) {
+      const preSelectedRowDetails = getDetailsOfPreSelectedRow();
+      const selectedRow = preSelectedRowDetails?.original ?? {};
+      const selectedRowId = preSelectedRowDetails?.id ?? null;
+      if (page.find((row) => row.id === selectedRowId && _.isEqual(selectedRow, row.original))) {
+        if (highlightSelectedRow) {
+          setExposedVariables({ selectedRow: selectedRow, selectedRowId: selectedRowId }).then(() => {
+            toggleRowSelected(selectedRowId, true);
+            mergeToTableDetails({ selectedRow: selectedRow, selectedRowId: selectedRowId });
+          });
+        } else {
+          toggleRowSelected(selectedRowId, true);
+        }
+      }
+    }
+    //hack : in the initial render, data is undefined since, upon feeding data to the table from some query, query inside current state is {}. Hence we added data in the dependency array, now question is should we add data or rows?
+  }, [JSON.stringify(defaultSelectedRow), JSON.stringify(data)]);
```

Comment:
I think page data is enough instead of complete table data unless if there is any edgecase